### PR TITLE
Add Outputs to test options

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Description

Right now if you want the terraform outputs you skip the teardown and get the outputs from the statefile with the code version of terraform outputs. This is not a great user experience and the outputs should be available in the TestOptions.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
- fix upgrade tests to now respect skipteardown and disable temp dir options
- added LastTestTerraformOutputs to options and tests
- added additional cleanup of terraform files in teardown, this is primarily for local development quality of life improvement
- removed un-necessary tearDowns

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: testhelper/tests.go [hunk 1]
```diff
@@ -228,6 +236,8 @@ func (options *TestOptions) testTearDown() {
 				logger.Log(options.Testing, "Deleting the temp working directory")
 				os.RemoveAll(options.baseTempWorkingDir)
 			}
+			//Clean up terraform files
+			CleanTerraformDir(options.TerraformDir)
```
Comment (toddgiguere) [inline testhelper/tests.go:240]: Will this be an issue if the TerraformDir was a temp dir that was removed a few lines up?

File: testhelper/terraform.go [hunk 2]
```diff
@@ -75,17 +76,46 @@ func ValidateTerraformOutputs(outputs map[string]interface{}, expectedKeys ...st
...
+			if err := os.RemoveAll(filePath); err != nil {
+				// Ignore errors, just log them
+				log.Printf("Error removing file %s: %s", fileName, err)
 			}
 		}
 	}
 }
+
+// Helper function to check if a string is in a slice of strings
+func contains(slice []string, item string) bool {
```
Comment (toddgiguere) [inline testhelper/terraform.go:114]: I think this function is a dupliate of `common/StrArrayContains()`: https://github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/blob/17605418d989134cac407be40c7f35bfa833019e/common/general.go#L153

**Review by toddgiguere — CHANGES_REQUESTED** [review]

File: testhelper/tests.go (hunk 1 above)
Comment (daniel-butler-irl) [reply testhelper/tests.go:240]: No its best effort, it will log an error but wont fail anything

**Review by toddgiguere — APPROVED** [review]

**Pull request merged**